pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -447,6 +451,7 @@ func generateObjectMeta(el *v1alpha1.EventListener) metav1.ObjectMeta {
 		Name:            el.Status.Configuration.GeneratedResourceName,
 		OwnerReferences: []metav1.OwnerReference{*el.GetOwnerReference()},
 		Labels:          mergeLabels(el.Labels, GenerateResourceLabels(el.Name)),
+		Annotations:     el.Annotations,
```

Comment:
I think we should merge any extra annotations that might have been added by something besides the EL controller instead of overwriting it (just like we are doing it with the labels above).

Reply:
@dibyom We are also overwriting in case of labels as per my understanding. We are merging labels to add the few labels which are generated by triggers controller, apart from that in case of labels also we are doing the same, copying the available in eventlistener, add the few generated by the controller and propagate to svc/deployment.

If we merge the extra annotation, don't do any anything which is already added, then how we can remove the one which a user did by mistake e.g. I have added an annotation with a typo in key. We will be in the same situation as before. So I think whatever the current set in EL should be copied further.

And as per my little understanding with the codebase, we are doing the same with labels. Please correct me here if wrong.

Reply:
@piyush-garg I agree with @dibyom that we shouldn't overwrite annotations which are added previously

> We are also overwriting in case of labels as per my understanding

Coming to labels with the current triggers code if you see here https://github.com/tektoncd/triggers/blob/master/pkg/reconciler/v1alpha1/eventlistener/eventlistener.go#L457-L459 we are not  overwriting instead we are merging 

> If we merge the extra annotation, don't do any anything which is already added, then how we can remove the one which a user did by mistake

I believe its responsibility of the user to pass correct annotations.
and we should merge annotations instead of overwriting same way like k8s `Deployment`

Reply:
Yes we are just copying labels coming from eventlistener and merging with labels which are generated by triggers controller.
For annotation we don't have any annotation which are generated by triggers controller so we should just copy whatever is coming from eventlistener yaml and imo we don't require any merging functionality here 

>I think we should merge any extra annotations that might have been added by something besides the EL controller instead of overwriting it

@dibyom  when you say `something besides the EL controller` do you mean annotation added by user ?

Reply:
Hey, sorry for the delayed response. 

@piyush-garg you are right -> we do overwrite the labels.  The use case I was thinking of is the following -- say someone currently has a custom annotation set directly on the EL service/deployment. Now, when they upgrade to Triggers containing this change, the custom annotation will be unset. So, they'd have to manually add their annotations to the EL in order to make it work. 

> If we merge the extra annotation, don't do any anything which is already added, then how we can remove the one which a user did by mistake e.g. I have added an annotation with a typo in key.

Yeah, that's not ideal either :( The user would have to delete them from the EL deployment/service manually.

It gets a bit more complicated if we consider [TEP-009](https://github.com/tektoncd/community/blob/master/teps/0008-support-knative-service-for-triggers-eventlistener-pod.md#design-details). In that design, users can specify annotations/labels directly for the EL deployment. So, we'll have to see if it makes sense to keep propagating them.

So, the options I see are:

1. Make the change in this PR and mark in the Release Notes that users have to manually update the EL's annotations if they added any custom annotations.

2. Merge any existing annotations on the labels/annotations with any applied on the EL instead of overwriting them. 

3. Wait for TEP-009 design to be implemented where users can directly specify a podTemplate with exactly the annotations they want for the deployment. ( We might want to augment that design with something like a serviceTemplate for the EL service as well)

Does that make sense? Thoughts @savitaashture @piyush-garg @khrm ?

Reply:
@dibyom 

Yes i agree with your point lets merge this PR with release note

> 3. Wait for TEP-009 design to be implemented where users can directly specify a podTemplate with exactly the annotations they want for the deployment. ( We might want to augment that design with something like a serviceTemplate for the EL service as well)

Yes need to rethink about the support of annotation for Service while implementing `TEP-009` as  that is bit tricky.
I will take care of this while implementation
Thank you for pointing it out here :+1: